clustermesh-apiserver/etcdinit/root.go

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package etcdinit
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"syscall"
+	"time"
+
+	"github.com/cilium/cilium/clustermesh-apiserver/etcdinit/defaults"
+	einit "github.com/cilium/cilium/pkg/etcd/init"
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+
+	"github.com/spf13/cobra"
+	client "go.etcd.io/etcd/client/v3"
+)
+
+var (
+	log                     = logging.DefaultLogger.WithField(logfields.LogSubsys, "etcdinit")
+	etcdBinaryLocation      = "/usr/bin/etcd"
```

Comment:
I wondered about this. If you ran this binary outside of it's container that would make sense. But inside the container there is no other etcd. I guess you could do something bizarre like mount a replacement etcd in a volume (😱). You could build an image atop this one, but I guess if you did you could just replace the etcd binary in the same path.

I can add it if you'd prefer, but I didn't see the value and worried about confusing people into thinking it's realistic to configure it.

Reply:
I don't have a strong opinion here, and I agree with your point. I just mentioned that possibility because it seemed a bit more generic, although realistically it should never be changed.

Reply:
I needed to make the logging and error handling better everywhere anyway. So I cleaned it up a bit. Unsure if it's better. 🤔